ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -548,3 +562,13 @@ private SocketChannel channel(SelectionKey key) {
     return (SocketChannel) key.channel();
   }
 }
+
+class PendingHandshakeTransmission {
```

Comment:
Also not sure why it *has* to be used in metrics classes. Usage in metrics classes shouldn't determine where actual classes are. Metrics could very well use `Selector` as the owning class and it would still make sense.